cargo-dist/src/announce.rs

```diff
@@ -357,117 +354,29 @@ pub(crate) fn select_tag(
             // like `init` or `generate` which just wants us to hand it everything
             // and doesn't care about coherent announcements. So use a fake tag
             // and hand it the fully unconstrained list of rust_releases.
-            announcing.tag = Some("v1.0.0-FAKEVER".to_owned());
+            announcing.tag = "v1.0.0-FAKEVER".to_owned();
             announcing.prerelease = true;
-            announcing.version = Some("1.0.0-FAKEVER".parse().unwrap());
+            announcing.release = ReleaseType::Version("1.0.0-FAKEVER".parse().unwrap());
         }
     }
+
+    let mut version = None;
+    let mut package = None;
+    match &announcing.release {
+        ReleaseType::Package(id) => package = Some(PackageIdx(*id)),
+        ReleaseType::Version(ver) => version = Some(ver.clone()),
+        ReleaseType::None => {}
+    }
+
     Ok(AnnouncementTag {
-        tag: announcing
-            .tag
-            .expect("integrity error: failed to select announcement tag"),
-        version: announcing.version,
-        package: announcing.package,
+        tag: announcing.tag,
+        version,
+        package,
```

Comment:
oh i see you unpack them back to the Evil format to minimize churn.

Reply:
Yeah, eventually we can do a refactor here where the upstream type becomes less evil, but this gets us unblocked today.